### DISCUSSION TITLE
Publishing Messages using management UI not working

### DISCUSSION THREAD

**Discussion by danammeansbear** [rabbitmq/rabbitmq-server#9380] [Other]
### Describe the bug

I am trying to resend some event messages using rabbitmq. When I log into the management UI and go to a queue and publish a message using the UI, they immediately are un-acked and eventually make it to the corresponding Dead letter queue. 

Do I need to modify my config file to allow for people to send messages over the UI? 

### Reproduction steps

1. fresh rabbitmq instal of 3.12.0 and erlang 25.3
2. allow for Local AD groups for authentication into the management UI 
3. publish a message manually. 
...

### Expected behavior

When we run our ETL's using c#, they run fine. When we try to publish a message manually using the UI, then we run into the issue of the messages immediately being un-acked. we expected them to be published and consumed right away.

**Comment by michaelklishin** [comment]
We cannot suggest much with just "allow for local AD groups for authentication", we won't guess as to what your LDAP rules and schema are are.

The user that you use over the management UI may have insufficient permissions for the virtual host in question. Your .NET clients may be using a different user.

Looking at the module in question, `rabbit_mgmt_wm_exchange_publish`, I don't see any permissions that such user would need beyond the standard [permissions for the `basic.publish` operation](https://rabbitmq.com/access-control.html).

Again, we are guessing since we have no no logs, LDAP queries or schema to reason about.

**Comment by michaelklishin** [comment]
I have one more guess. The HTTP API publishing uses `mandatory = true` (you can search for "mandatory" in the [Publishers guide](https://rabbitmq.com/publishers.html#unroutable), for example). Your clients may be publishing messages as non-mandatory.

The topology may be set up to use alternate exchanges, and so on. The delayed message exchange plugin does not support mandatory routing at all, by design.

**Comment by danammeansbear** [comment]
I verified that the user has the same permissions as I do. 
The c# clients user can send a message to the service bus and it be consumed as expected. 
The c# clients user cannot send that same payload onto the service bus and be consumed as expected when using the RMQ management UI. 

when I publish a message using the UI. this shows up in the log. 
```
2023-09-12 14:39:20.553000-04:00 [info] <0.386.0>     LDAP DECISION: does adabdoub have tag administrator? true
2023-09-12 14:39:20.553000-04:00 [info] <0.12737.16> LDAP DECISION: login for adabdoub: ok

```

**Reply by lukebakken** [reply]
> The c# clients user can send a message to the service bus and it be consumed as expected. The c# clients user cannot send that same payload onto the service bus and be consumed as expected when using the RMQ management UI.

I know this sounds obvious, but there must be a difference in how the messages are being published. My _guess_ is that you are not using the same routing key in both scenarios.

The _only_ way we can continue to assist you is if you export your definitions and provide _exact_ instructions for how to reproduce the problem.

**Comment by danammeansbear** [comment]
So I started off by examining the messages that were being sent over using the c# software, I noticed that there are message properties.
1)Ill find a message that hasnt been consumed and copy it down somewhere since  I know that message is going to work and be consumed. 
2)Ill send a message over the UI without these properties filled in and they are marked unroutable and thus get sent to are Dead Letter queue. 
3) I did the same but filled in all properties and they were marked unacked but were eventually consumed by the consumer and never sent to the dead letter queue.


I don think that in previous version of rabbitmq that this needed to be specified. I also believe that in previous version of rabbitmq, we were able to specify the routing key we want to send the message over to on a specific queue. This option is now only available when I go to the Common Exchange. that holds all queues and bindings.

**Reply by danammeansbear** [reply]
I am going to keep testing before I close this issue out.
